Sources/Orbit/Components/Button.swift

```diff
@@ -147,152 +145,183 @@ public extension Button {
     ///   - style: A visual style of component. A `status` style can be optionally modified using `status()` modifier when `nil` value is provided.
     init(
         _ label: String,
-        icon: Icon.Content? = nil,
-        disclosureIcon: Icon.Content? = nil,
-        style: Style = .primary,
-        size: Size = .default,
+        icon: Icon.Symbol? = nil,
+        disclosureIcon: Icon.Symbol? = nil,
+        type: ButtonType = .primary,
+        size: ButtonSize = .default,
         action: @escaping () -> Void
-    ) {
-        self.label = label
-        self.icon = icon
-        self.disclosureIcon = disclosureIcon
-        self.style = style
-        self.size = size
-        self.action = action
+    ) where LeadingIcon == Icon, TrailingIcon == Icon {
+        self.init(
+            label,
+            type: type,
+            size: size,
+            action: action
+        ) {
+            Icon(icon, size: size.textSize.iconSize)
+        } disclosureIcon: {
+            Icon(disclosureIcon, size: size.textSize.iconSize)
+        }
     }
 
     /// Creates Orbit Button component with icon only.
     ///
     /// - Parameters:
     ///   - style: A visual style of component. A `status` style can be optionally modified using `status()` modifier when `nil` value is provided.
     init(
-        _ icon: Icon.Content,
-        style: Style = .primary,
-        size: Size = .default,
+        _ icon: Icon.Symbol,
+        type: ButtonType = .primary,
+        size: ButtonSize = .default,
         action: @escaping () -> Void
-    ) {
+    ) where LeadingIcon == Icon, TrailingIcon == EmptyView {
```

Comment:
Sure, but we could have the `label` default to `""`. Which could lead to "empty" buttons, which is what I originally asked about.

Reply:
True, but we would still need to call it as `Button(icon: .grid)` and not `Button(.grid)`, which was the original reason to add this init before. We could remove it, but it may be useful to have this init to be sure this is a variant specifically for icon-only, so the call site does not have to wonder what would happen when provided the empty label value.